### PR TITLE
[DIR-1378] Disabling the Run Button in the header on unsaved Changes

### PR DESCRIPTION
**Pull request by julia-zielke**
## Description

When there are unsaved changes in the Workflow Editor, the second Run Button (the color filled one in the header) is now disabled too.

## Checklist

- [x] Documentation updated if required
- [x] Test coverage is appropriate
      
## Checklist Internal

- [x] Linear issue linked (e.g. [DIR-XXXX] pull request title)
- [x] Has the PR been labeled

### CODE REVIEW

**Comment by julia-zielke** [comment]
@stefan-kracht 
Thanks again for the explanation about useContext. I feel enlightened!

I renamed everything from runable to disabled because I did not want it to be:
`<Button disabled={!runable}>`

**Review by stefan-kracht — CHANGES_REQUESTED** [review]
Nice. Great that you added some E2E tests as well. Made some comments below.

File: ui/src/pages/namespace/Explorer/Workflow/Edit/WorkflowEditor.tsx [hunk 1]
```diff
@@ -39,13 +41,15 @@ const WorkflowEditor: FC<{
...
       setHasUnsavedChanges(false);
+      setDisabled(false);
     },
   });
 
   const [editorContent, setEditorContent] = useState(workflowDataFromServer);
 
   const onEditorContentUpdate = (newData: string) => {
     setHasUnsavedChanges(workflowDataFromServer !== newData);
+    setDisabled(true);
```
[the remark this answers is not in the document]
Comment (julia-zielke) [reply ui/src/pages/namespace/Explorer/Workflow/Edit/WorkflowEditor.tsx:52]: Oh yes, thanks!

**Review by stefan-kracht — APPROVED** [review]
Looks 👌

**Pull request merged**